rotkehlchen/db/history_events.py

```diff
@@ -227,17 +254,27 @@ def get_history_events(
         """
         query, bindings = filter_query.prepare()
 
-        if has_premium:
-            query = 'SELECT * from history_events ' + query
-            cursor.execute(query, bindings)
+        if isinstance(filter_query, EvmEventFilterQuery):
+            # Select only evm events if `EvmEventFilterQuery` is used
+            join_query = EVM_EVENT_JOIN
         else:
-            query = 'SELECT * FROM (SELECT * from history_events ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) ' + query  # noqa: E501
-            cursor.execute(query, [FREE_HISTORY_EVENTS_LIMIT] + bindings)
+            join_query = ALL_EVENTS_DATA_JOIN
 
+        base_query = f'SELECT {HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS} {join_query} '
+        if has_premium is False:
+            base_query += f'({join_query} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '  # noqa: E501
+            bindings.insert(0, FREE_HISTORY_EVENTS_LIMIT)
+
+        cursor.execute(base_query + query, bindings)
         output = []
         for entry in cursor:
             try:
-                deserialized = HistoryBaseEntry.deserialize_from_db(entry)
+                # Deserialize event depending on its type
+                deserialized: Union[HistoryBaseEntry, EvmEvent]
+                if determine_event_type(Location.deserialize_from_db(entry[4])) == 'evm_event':
```

Comment:
I believe you had a discussion already with lefteris about the usage of constants like this but don't know what was the output of that.

Here I believe that is better to use enums for different reasons:

1. The constants are used in different places in the code and they are not imported but written as string.
2. We  will introduce more probably in the future and this complicates things
3. Is easier to navigate and search docs if this is a enum instead of a string constant

Reply:
Okay, will change to an enum.